docs/proposal/20211207-delegated-storage.md

```diff
@@ -536,42 +574,59 @@ Capact Local Hub calls proper storage backend service while accessing the TypeIn
     }
     
     message OnLockRequest {
-      string typeinstance_id = 1;
+      string type_instance_id = 1;
       bytes context = 2;
       string locked_by = 3;
     }
     
     message OnLockResponse {}
     
     message OnUnlockRequest {
-      string typeinstance_id = 1;
+      string type_instance_id = 1;
       bytes context = 2;
     }
     
     message OnUnlockResponse {}
     
     // services
     
-    service StorageBackend {
+    // ValueAndContextStorageBackend handles the full lifecycle of the TypeInstance.
+    // TypeInstance value is always provided as a part of request. Context may be provided but it is not required.
+    service ValueAndContextStorageBackend {
       // value
       rpc GetValue(GetValueRequest) returns (GetValueResponse);
+      rpc OnCreate(OnCreateValueAndContextRequest) returns (OnCreateResponse);
+      rpc OnUpdate(OnUpdateValueAndContextRequest) returns (OnUpdateResponse);
+      rpc OnDelete(OnDeleteValueAndContextRequest) returns (OnDeleteResponse);
+    
+      // lock
+      rpc GetLockedBy(GetLockedByRequest) returns (GetLockedByResponse);
+      rpc OnLock(OnLockRequest) returns (OnLockResponse);
+      rpc OnUnlock(OnUnlockRequest) returns (OnUnlockResponse);
+    }
+    
+    // ContextOnlyStorageBackend handles TypeInstance lifecycle based on the context, which is required. TypeInstance value is never passed in input arguments.
```

Comment:
IMO there is no need to say `Only` the:
- `ValueAndContextStorageBackend`
- and `ContextStorageBackend`
names, are more consistent.